R/queue.R

```diff
@@ -62,6 +64,29 @@ Queue <- R6::R6Class("Queue", #nolint
       rrq::rrq_worker_len(self$controller)
     },
 
+    #' @description
+    #' Gets status of packet run
+    #'
+    #' @param job_id Id of redis queue job to get status of.
+    #' @return status of redis queue job
+    get_status = function(job_id) {
+      if (!rrq::rrq_task_exists(job_id, controller = self$controller)) {
+        porcelain::porcelain_stop("Job ID does not exist")
+      }
+      status <- rrq::rrq_task_status(job_id, controller = self$controller)
+      times <- rrq::rrq_task_times(job_id, controller = self$controller)
+
+      list(
+        status = scalar(status),
+        queue_position = if (status == "PENDING") scalar(rrq::rrq_task_position(job_id, controller = self$controller)) else NULL,
+        time_queued = scalar(times[1]),
+        time_started = if (!is.na(times[2])) scalar(times[2]) else NULL,
+        time_complete = if (!is.na(times[3])) scalar(times[3]) else NULL,
```

Comment:
i think jsonlite (package we use for serialising to json) converts NA to null so i think will be fine
```suggestion
        time_started = scalar(times[2]),
        time_complete = scalar(times[3]),
```
if you want to test how something will look in json form you can run `jsonlite::toJSON` on it